File: ProjectGray/ProjectGray/UnitStats.m
//
//  UnitStats.m
//  ProjectGray
//
//  Created by Matthew Ku on 2015-02-23.
//  Copyright (c) 2015 Tim Wang. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "UnitStats.h"

@interface UnitStats ()

@end

@implementation UnitStats

-(instancetype) init {
    //Stats for light ship type
    _lightShip.hull = 10;
    _lightShip.attRange = 2;
    _lightShip.damage = 2;
    _lightShip.moveRange = 2;
    _lightShip.accuracy = 100.0f;
    _lightShip.critChance = 25.0f;
    _lightShip.critModifier = 1.5f;
    _lightShip.actionsPerTurn = 3;
    _lightShip.engineHealth = 5;
    _lightShip.weaponHealth = 5.0f;
    _lightShip.shipHealth = 10;
    //Stats for medium ship type
    _medShip.hull = 10;
    _medShip.attRange = 2;
    _medShip.damage = 2;
    _medShip.moveRange = 2;
    _medShip.accuracy = 100.0f;
    _medShip.critChance = 25.0f;
    _medShip.critModifier = 1.5f;
    _medShip.actionsPerTurn = 3;
    _medShip.engineHealth = 5;
    _medShip.weaponHealth = 5.0f;
    _medShip.shipHealth = 10;
    //Stats for heavy ship type
    _heavyShip.hull = 10;
    _heavyShip.attRange = 2;
    _heavyShip.damage = 2;
    _heavyShip.moveRange = 2;
    _heavyShip.accuracy = 100.0f;
    _heavyShip.critChance = 25.0f;
    _heavyShip.critModifier = 1.5f;
    _heavyShip.actionsPerTurn = 3;
    _heavyShip.engineHealth = 5;
    _heavyShip.weaponHealth = 5.0f;
    _heavyShip.shipHealth = 10;
    return self;
}
@end

File: ProjectGray/ProjectGray/UnitStats.c
//
//  UnitStats.c
//  ProjectGray
//
//  Created by Shane Spoor on 2015-02-23.
//  Copyright (c) 2015 Tim Wang. All rights reserved.
//

#include "UnitStats.h"

const char* factionNames[] =
{
    "Vikings",
    "Aliens"
};

const char* shipImages[NUM_FACTIONS][NUM_CLASSES] =
{
    //vikings
    {
        "l_viking",
        "m_viking",
        "h_viking"
    },
    //aliens
    {
        "l_gray",
        "m_gray",
        "h_gray"
    }

};

const ShipStats factionShipStats[NUM_FACTIONS][NUM_CLASSES] = {
    // VIKINGS
    {
        // Light ship base stats
        {
            3,          // Attack range
            3,          // Heal range
            2,          // Action points per attack
            2,          // Action points per heal
            1,          // Moves (tiles) per action point
            4,          // Action point pool (number at the beginning of each turn)
            0.70f,      // Accuracy
            0.10f,      // Critical hit chance
            1.5f,       // Critical hit modifier
            15,         // Damage modifier
            0.7f,       // Hull
            5,          // Engine health
            1.0f,       // Weapon health
            100         // Ship health
        },
        // Medium ship base stats
        {
            3,          // Attack range
            3,          // Heal range
            2,          // Action points per attack
            2,          // Action points per heal
            1,          // Moves (tiles) per action point
            3,          // Action point pool (number at the beginning of each turn)
            0.70f,      // Accuracy
            0.10f,      // Critical hit chance
            2.0f,      // Critical hit modifier
            5,          // Damage modifier
            0.12f,         // Hull
            5,          // Engine health
            1.0f,       // Weapon health
            100         // Ship health
        },
        // Heavy ship base stats
        {
            1,          // Attack range
            2,          // Heal range
            2,          // Action points per attack
            2,          // Action points per heal
            1,          // Moves (tiles) per action point
            2,          // Action point pool (number at the beginning of each turn)
            0.70f,      // Accuracy
            0.20f,      // Critical hit chance
            1.25f,       // Critical hit modifier
            30,          // Damage modifier
            0.25f,         // Hull
            5,          // Engine health
            1.0f,       // Weapon health
            100         // Ship health
        },
    },
    // ALIENS
    {
        // Light ship base stats
        {
            3,          // Attack range
            3,          // Heal range
            2,          // Action points per attack
            2,          // Action points per heal
            1,          // Moves (tiles) per action point
            4,          // Action point pool (number at the beginning of each turn)
            0.70f,      // Accuracy
            0.10f,      // Critical hit chance
            1.5f,       // Critical hit modifier
            15,         // Damage modifier
            0.7f,          // Hull
            5,          // Engine health
            1.0f,       // Weapon health
            100         // Ship health
        },
        // Medium ship base stats
        {
            3,          // Attack range
            3,          // Heal range
            2,          // Action points per attack
            2,          // Action points per heal
            1,          // Moves (tiles) per action point
            3,          // Action point pool (number at the beginning of each turn)
            0.70f,      // Accuracy
            0.10f,      // Critical hit chance
            2.0f,      // Critical hit modifier
            5,          // Damage modifier
            0.12f,         // Hull
            5,          // Engine health
            1.0f,       // Weapon health
            100         // Ship health
        },
        // Heavy ship base stats
        {
            1,          // Attack range
            2,          // Heal range
            2,          // Action points per attack
            2,          // Action points per heal
            1,          // Moves (tiles) per action point
            2,          // Action point pool (number at the beginning of each turn)
            0.70f,      // Accuracy
            0.20f,      // Critical hit chance
            1.25f,       // Critical hit modifier
            30,          // Damage modifier
            0.25f,         // Hull
            5,          // Engine health
            1.0f,       // Weapon health
            100         // Ship health
        },
    }
};